esp-hal/src/dma/buffers.rs

```diff
@@ -4,25 +4,252 @@ use core::{
+        #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+        #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+        pub enum BurstConfig {
+            /// Burst mode is disabled.
+            Disabled,
+
+            /// Burst mode is enabled.
+            Enabled,
+        }
+
+        impl BurstConfig {
+            /// The default burst mode configuration.
+            pub const DEFAULT: Self = Self::Disabled;
+        }
+
+        impl Default for BurstConfig {
+            fn default() -> Self {
+                Self::DEFAULT
+            }
+        }
 
-    /// Burst mode is enabled.
-    Enabled,
+        type InternalBurstTransfer = BurstConfig;
+    }
 }
 
-impl BurstConfig {
+#[cfg(psram_dma)]
+impl ExternalBurstSize {
+    const fn min_psram_alignment(self, direction: TransferDirection) -> usize {
+        // S2: Specifically, size and buffer address pointer in receive descriptors
+        // should be 16-byte, 32-byte or 64-byte aligned. For data frame whose
+        // length is not a multiple of 16 bytes, 32 bytes, or 64 bytes, EDMA adds
+        // padding bytes to the end.
+
+        // S3: Size and Address for IN transfers must be block aligned. For receive
+        // descriptors, if the data length received are not aligned with block size,
+        // GDMA will pad the data received with 0 until they are aligned to
+        // initiate burst transfer. You can read the length field in receive descriptors
+        // to obtain the length of valid data received
+        if matches!(direction, TransferDirection::In) {
+            self as usize
+        } else {
+            // S2: Size, length and buffer address pointer in transmit descriptors are not
+            // necessarily aligned with block size.
+
+            // S3: Size, length, and buffer address pointer in transmit descriptors do not
+            // need to be aligned.
+            1
+        }
+    }
+}
+
+impl InternalBurstTransfer {
     pub(super) fn is_burst_enabled(self) -> bool {
         !matches!(self, Self::Disabled)
     }
+
+    const fn min_dram_alignment(self, direction: TransferDirection) -> usize {
+        // IN transfers must be word aligned
+        if matches!(direction, TransferDirection::In) {
+            4
+        } else {
+            // OUT transfers have no alignment requirements, except for ESP32, which is
+            // described below.
+            if cfg!(esp32) {
+                // SPI DMA: Burst transmission is supported. The data size for
+                // a single transfer must be four bytes aligned.
+                // I2S DMA: Burst transfer is supported. However, unlike the
+                // SPI DMA channels, the data size for a single transfer is
+                // one word, or four bytes.
+                4
+            } else {
+                1
+            }
+        }
+    }
+}
+
+const fn max(a: usize, b: usize) -> usize {
+    if a > b {
+        a
+    } else {
+        b
+    }
+}
```

Comment:
The standard library provides this. Or is `const` the problem?

Reply:
the Ord trait isn't const